### DISCUSSION TITLE
限制文生图 图片的展示尺寸

### DISCUSSION THREAD

**Discussion by echoniuniu** [lobehub/lobehub#1413] [General]
当我在使用文生图时候他返回图片是如下这样的 ，感觉展示样式太大了，在聊天窗口中不能直接观察到图片整体

<img src="https://github.com/lobehub/lobe-chat/assets/145311948/7ac2a95d-f224-4d8b-86ec-c80a9f14c794" width="400">

而且程序UI本身提供了，如下大图预览功能
<img src="https://github.com/lobehub/lobe-chat/assets/145311948/f5563a4f-d8fa-46b4-8ed4-d19c26023eff" width="400">

下面是我手动使用 css 样式调节的

希望能够像下面这样限制图片展示尺寸，**相对于当前聊天对话框窗口，能够直接的看到图片的整体**

![image](https://github.com/lobehub/lobe-chat/assets/145311948/ecc7ea6a-109d-42ae-a68e-7b6a00a0ddbc)

**Comment by arvinxx** [comment]
这个实现上感觉不太行。不太可能动态感知窗口尺寸来调整大小的

**Comment by echoniuniu** [comment]
请理解,我并不特别清楚 当前使用什么UI框架开发的,

我通过前端工具 找到一个窗体,通过监听他可以动态改变图片尺寸


![image](https://github.com/lobehub/lobe-chat/assets/71559822/56058c36-abbb-4b47-8517-cdd52fe87b45)

我手写了下面一个脚本 来 完成这件事, 供参考
```js

const targetElement = document.querySelector('.acss-9v0z6m');
// 因为是前端工具 渲染的,这个窗体好像没有一个实在的类名
const resizeObserver = new ResizeObserver(entries => {
    for (let entry of entries) {
        if (entry.target === targetElement) {
            const newHeight = entry.contentRect.height;
            const newWidth = entry.contentRect.width;
            let visMin = Math.min(newHeight, newWidth) * 0.5;
            const imageElements = document.querySelectorAll('.ant-image');
            imageElements.forEach(imageElement => {
                imageElement.style.maxWidth = `${visMin} px`;
                imageElement.style.maxHeight = `${visMin} px`;
                const imageChildElements = Array.from(imageElement.children);
                imageChildElements.forEach(imageChildElement => {
                    imageChildElement.style.maxWidth = `${visMin}px`;
                    imageChildElement.style.maxHeight = `${visMin}px`;
                });
            });
        }
    }
});

resizeObserver.observe(targetElement);

```
我不是很会写js, 不过外挂js 脚本也是件很不优雅的事

https://github.com/lobehub/lobe-chat/assets/71559822/d21c28b9-3d69-4579-95d2-f00856da29f6

**Reply by echoniuniu** [reply]
可能我描述不太清楚,就是下面箭头这一部分 所谓"聊天窗口" 

![image](https://github.com/lobehub/lobe-chat/assets/145311948/e70cde7b-84d1-4595-9f69-9c592136fd76)

**Comment by arvinxx** [comment]
我个人觉得后面可以给个配置来设定默认的图片大小，目前是写死的。但是按视窗进行动态化缩放，这个应该不计划做的
